django/views/generic/list.py

```diff
@@ -181,7 +181,13 @@ def get_template_names(self):
         if hasattr(self.object_list, 'model'):
             opts = self.object_list.model._meta
             names.append("%s/%s%s.html" % (opts.app_label, opts.model_name, self.template_name_suffix))
-
+        else:
+            if self.object_list is None and not names:
+                raise ImproperlyConfigured(
```

Comment:
Not sure whether this is the right exception - should be `TypeError`.

Reply:
Good point. I took my cue from the error raised in get_queryset above, which does raise ImproperlyConfigured. I think TypeError makes more sense, will change.

Reply:
Changed my mind about the TypeError, see comment above ;)

Reply:
Fair enough if there is precedence for it. I thought `ImproperlyConfigured` was usually used for settings or something affected by a setting being incorrect.